src/bodypart.cpp

```diff
@@ -2,13 +2,57 @@
 #include "translations.h"
 #include "rng.h"
 #include "debug.h"
+#include "generic_factory.h"
 #include <map>
 #include <unordered_map>
 
+int body_part_struct::size_sum = 0;
+
+side opposite_side( side s )
+{
+    switch( s ) {
+        case side::BOTH:
+            return side::BOTH;
+        case side::LEFT:
+            return side::RIGHT;
+        case side::RIGHT:
+            return side::LEFT;
+    }
+
+    return s;
+}
+
+namespace io
+{
+
+static const std::map<std::string, side> side_map = {{
+        { "left", side::LEFT },
+        { "right", side::RIGHT },
+        { "both", side::BOTH }
+    }
+};
+
+template<>
+side string_to_enum<side>( const std::string &data )
+{
+    return string_to_enum_look_up( side_map, data );
+}
+
+}
+
+static std::array < body_part_struct, num_bp + 1 > human_anatomy;
+static std::unordered_map<std::string, body_part> body_parts_map;
```

Comment:
Doesn't it require a proper string id to work? At the moment the "id" of body part is an enum, so I'd rather implement the generic factory parts once I can get rid of the fake IDs.